src/Compilers/Test/Core/Diagnostics/DiagnosticDescription.cs

```diff
@@ -444,6 +461,29 @@ public override string ToString()
             return sb.ToString();
         }
 
+        private static ImmutableArray<string> GetFormatSpecifiers(string messageFormat)
+        {
+            ImmutableArray<string> specifiers = default;
+            if (Regex.Matches(messageFormat, @"\{\d+(:\d+)?\}") is { Count: > 0 } matches)
+            {
+                var builder = ArrayBuilder<string>.GetInstance();
+                foreach (Match match in matches)
+                {
+                    if (match.Groups.Count > 1 && match.Groups[1].Success)
+                    {
+                        builder.Add($@"{{0{match.Groups[1].Value}}}");
```

Comment:
Shouldn't we be extracting the position specifier? IE, it shouldn't be `0<format specifier from the original>`, it should be `<position specifier from the original><format specifier from the original>`? I think a comment explanation here would be useful.

Reply:
Ah yes I should add an explanation here because it is confusing. Since the formatting happens argument by argument, we are only ever using a single specifier at once so we need to use `0` for the position specifier